xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/java/org/xwiki/notifications/filters/migration/R160500000XWIKI22271DataMigration.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.xwiki.notifications.filters.migration;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.notifications.filters.internal.DefaultNotificationFilterPreference;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+import org.xwiki.wiki.descriptor.WikiDescriptorManager;
+import org.xwiki.wiki.manager.WikiManagerException;
+
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.store.migration.DataMigrationException;
+import com.xpn.xwiki.store.migration.XWikiDBVersion;
+import com.xpn.xwiki.store.migration.hibernate.AbstractHibernateDataMigration;
+
+/**
+ * Ensure to clean up filters related to deleted wikis.
+ *
+ * @version $Id$
+ * @since 16.5.0
+ * @since 16.4.1
+ */
+@Component
+@Named("R160500000XWIKI22271")
+@Singleton
+public class R160500000XWIKI22271DataMigration extends AbstractHibernateDataMigration
+{
+    private static final String SEARCH_FILTERS_STATEMENT = "select nfp "
+        + "from DefaultNotificationFilterPreference nfp "
+        + "where nfp.id not in ("
+        + "select nfp2.id from DefaultNotificationFilterPreference nfp2 "
+        + "where nfp2.page like :wikiPrefix or "
+        + "nfp2.pageOnly like :wikiPrefix or "
+        + "nfp2.user like :wikiPrefix or "
+        + "nfp2.wiki = :wikiId"
+        + ")";
+
+    private static final String DELETE_FILTER_STATEMENT = "delete from DefaultNotificationFilterPreference "
+        + "where internalId in (:filterIds)";
+
+    private static final int BATCH_SIZE = 100;
```

Comment:
isn't 100 a bit small?

Reply:
That's what I used in the migration for moving the filters, see: https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/java/org/xwiki/notifications/filters/migration/R160100000XWIKI21738DataMigration.java#L67

Honestly I've no idea what's best value here.

Reply:
After reading where it's used, 100 looks ok, but it's not obvious when reading the value out of context.